Enable main.py to be run locally

Ists-Cilveks opened the pull request:
Using a python interpreter to run main_locally.py is drastically faster (for me) than using pyscript every time.
Also, launch.json makes this even more convenient in VSCode.

Ists-Cilveks committed:
Enable main,py to be run locally

Using a python interpreter to run main_locally.py is drastically faster (for me) than using pyscript every time.
Also, launch.json makes this even more convenient in VSCode.

Ists-Cilveks committed:
Enable main,py to be run locally

Using a python interpreter to run main_locally.py is drastically faster (for me) than using pyscript every time.
Also, launch.json makes this even more convenient in VSCode.

saizai commented on renderer/pyscript-offline/public/.vscode/launch.json:
```diff
@@ -0,0 +1,22 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
// comments are prohibited in JSON

saizai commented:
could you please pull / rebase this from master?

Ists-Cilveks committed:
Remove comments from launch.json

Signed-off-by: jan Talisoso <38190993+Ists-Cilveks@users.noreply.github.com>

Ists-Cilveks committed:
Remove comments from launch.json

Signed-off-by: jan Talisoso <38190993+Ists-Cilveks@users.noreply.github.com>

Ists-Cilveks committed:
Temporarily rename question mark file bc I can't delete it

Signed-off-by: jan Talisoso <38190993+Ists-Cilveks@users.noreply.github.com>

Ists-Cilveks committed:
Delete question mark file

There's already a version with no "?" or "2" at the end.

Ists-Cilveks committed:
Enable multiple texts on one page

Making main.py into a class allows it to have multiple methods that main_locally and main_via_pyscript can use repeatedly without needing to pass lots of arguments.

Ists-Cilveks committed:
Enable multiple texts on one page

Making main.py into a class allows it to have multiple methods that main_locally and main_via_pyscript can use repeatedly without needing to pass lots of arguments.

Ists-Cilveks committed:
Merge branch 'run-renderer-locally' of https://github.com/saizai/unlws into run-renderer-locally

Ists-Cilveks committed:
Bring back the example with different penalties

This example shows the result of using different penalty functions in the relaxer.
It was merged into master in commit 41d4aeadebee43d0f71f04f96e328353398acabd but I removed it while rebasing. Maybe I could've handled it better, but now it's a separate commit.

Ists-Cilveks committed:
Base the local html on index.html

This way style and other bits added to index.html will be reflected both when running locally and with pyscript. But currently it assumes a specific template for the body.

I needed to make the <meta /> and <link /> tags in index.html self closing for the minidom parser to work. Unfortunately, this style will have to be followed in all future versions of index for the parser to keep working.

Ists-Cilveks committed:
Get rid of redundant whitespace in output.html

Ists-Cilveks committed:
Make subclasses of Main instead of passing methods to it

Gives Main's methods access to "self", but I mainly just wanted to not repeat the docstring in main_locally and main_via_pyscript, and to get hints for what arguments Main's unimplemented methods need.

Ists-Cilveks committed:
Fix newlines in the output

render_with_comments needs `\n`, not `<br>`
